TFT/src/User/Fatfs/myfatfs.c

```diff
@@ -4,6 +4,42 @@
 
 FATFS fatfs[FF_VOLUMES];  // FATFS work area
 
+/**
+ * @brief  Compare file/folder details according to sort settings
+ *
+ * @param name1 name of first file/folder
+ * @param date1 date/time for first file/folder
+ * @param name2 name of second file/folder
+ * @param date2 date/time for second file/folder
+ */
+bool compareFile(char * name1, int32_t date1, char * name2, int32_t date2)
+{
+  // sort by date
+  if (infoSettings.files_sort_by <= SORT_DATE_OLD_FIRST)
+  {
+    // file with most recent date displays first in newest first and last in oldest first
+    return ((date1 > date2) == infoSettings.files_sort_by % 2);
+  }
+  // sort by name
+  else
+  {
+    uint16_t maxlen = (strlen(name1) < strlen(name2)) ? strlen(name1) : strlen(name2);
+
+    // compare each character
+    for (uint16_t i = 0; i < maxlen; i++)
+    {
+      // convert all upper case characters to lower case
+      char a = (name1[i] > 64 && name1[i] < 91) ? (name1[i] + 32) : name1[i];
+      char b = (name2[i] > 64 && name2[i] < 91) ? (name2[i] + 32) : name2[i];
+
+      if (a != b)
+        return ((a < b) == infoSettings.files_sort_by % 2);
+    }
+    // file with longer name displays last in ascending order and first in descending order
+    return ((strlen(name1) < strlen(name2)) == infoSettings.files_sort_by % 2);
+  }
+}
+
```

Comment:
@digant73  Please merge commits to your branch by rebase. It looks like you are manually adding changes instead of rebasing it.

Reply:
the problem is that rebase doesn't work on my totoise git

Reply:
There is a rebase oprion in tortoise git. https://tortoisegit.org/docs/tortoisegit/tgit-dug-rebase.html
 And there is a rebase option in github desktop app also under Branch menu.

Reply:
I know, but it doesn't work properly. It mixed very old chenges instead of the last ones.